### ISSUE TITLE
Exceptions in `filter` should (optionally) be re-raised

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#7465]
We had 2 cases of this within 24 hours in the IUC gitter by experienced tool writers (xref #7464). Currently an exception is interpreted as `False`, so outputs will be created if an invalid filter tag is used for filtering an output. Just failing outright would probably help tool devs and possibly increase tool quality overall.

**Comment by pvanheus** [comment]
+1 - except the exception is interpreted as `True`.

**Comment by mvdbeek** [comment]
https://github.com/natefoo/galaxy/blob/dev/lib/galaxy/tools/actions/__init__.py#L921

**Comment by nsoranzo** [comment]
@mvdbeek The method returns the opposite of the boolean value of the filter.

**Comment by mvdbeek** [comment]
Isn't that what I said above ?

**Comment by nsoranzo** [comment]
@mvdbeek What we mean is that:
- For `<filter>True</filter>`, `filter_output()` returns `False` and the output dataset is created
- For `<filter>False</filter>`, `filter_output()` returns `True` and the output dataset is not created
- For `<filter>raise Exception()</filter>`, `filter_output()` returns `False` and the output dataset is created

So, a filter that raise an exception is equivalent to a filter that evaluates to `True`.

**Comment by mvdbeek** [comment]
Yes, in the opening post I'm talking about the `filter` function, not the filter ... which sounds as confusing as it is 😆

**Comment by wm75** [comment]
I guess this is still open and is the cause also for the unexpected behavior of filters for multiple, optional outputs like in https://github.com/galaxyproject/tools-iuc/pull/3283/files#r527870969 ?
My assumption is that if there's no option selected for the select with multiple=True the value is None and then in the data filter boils down to sth like: `"bed" in None`, which causes a TypeError, which results in the output to be generated?

If that's the case, then this is an issue that definitely deserves fixing. @bebatut just fell into this trap again in https://github.com/galaxyproject/tools-iuc/pull/4692.
